File: ruby/samples/get_options_prices.rb
# Load the gem
require 'intrinio-sdk'

# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR API KEY'
end

options_api = Intrinio::OptionsApi.new

identifier = "AAPL190318C00300510" # String | The Intrinio ID or code of the options contract to request prices for.

opts = { 
  start_date: nil, # String | Return option contract prices on or after this date.
  end_date: nil, # String | Return option contract prices on or before this date.
  page_size: 100, # Float | The number of results to return
  next_page: nil # String | Gets the next page of data from a previous API call
}

begin
  result = options_api.get_options_prices(identifier, opts)
  option = result.option
  prices = result.prices

  puts "*** OPTION ***"
  puts "ID:         #{option.id}"
  puts "Code:       #{option.code}"
  puts "Ticker:     #{option.ticker}"
  puts "Expiration: #{option.expiration}"
  puts "Strike:     #{option.strike}"
  puts "Type:       #{option.type}"

  puts ""
  puts "#{prices.size} prices found for #{identifier}"
  puts "--------------------------------------------------------------------------------"

  prices.each do |price|
    puts ""
    puts "Date:                      #{price.date}"
    puts "Close:                     #{price.close}"
    puts "Close bid:                 #{price.close_bid}"
    puts "Close ask:                 #{price.close_ask}"
    puts "Volume:                    #{price.volume}"
    puts "Volume bid:                #{price.volume_bid}"
    puts "Volume ask:                #{price.volume_ask}"
    puts "Trades:                    #{price.trades}"
    puts "Open interest:             #{price.open_interest}"
    puts "Open interest change:      #{price.open_interest_change}"
    puts "Next day open interest:    #{price.next_day_open_interest}"
    puts "Implied volatility:        #{price.implied_volatility}"
    puts "Implied volatility change: #{price.implied_volatility_change}"
    puts "Delta:                     #{price.delta}"
  end
rescue Intrinio::ApiError => e
  puts "Exception when calling OptionsApi->get_options_prices: #{e}"
end

File: javascript-node/samples/getOptionsPrices.js
var intrinioSDK = require('intrinio-sdk');
intrinioSDK.ApiClient.instance.authentications['ApiKeyAuth'].apiKey = "YOUR API KEY";
 
var optionsAPI = new intrinioSDK.OptionsApi();
 
var identifier = "AAPL190318C00300510"; // String | The Intrinio ID or code of the options contract to request prices for.
 
var opts = {};
 
optionsAPI.getOptionsPrices(identifier, opts).then(function(data) {
  var option = data.option;
  var prices = data.prices;
  
  console.log("*** OPTION ***");
  console.log("ID:         " + option.id);
  console.log("Code:       " + option.code);
  console.log("Ticker:     " + option.ticker);
  console.log("Expiration: " + option.expiration);
  console.log("Strike:     " + option.strike);
  console.log("Type:       " + option.type);
  
  console.log("");
  console.log(prices.length + ' options prices found for ' + identifier + '!');
  console.log("------------------------------------------------------------");
  
  prices.forEach(function(price) {
    console.log("");
    console.log("Date:                      " + price.date);
    console.log("Close:                     " + price.close);
    console.log("Close bid:                 " + price.close_bid);
    console.log("Close ask:                 " + price.close_ask);
    console.log("Volume:                    " + price.volume);
    console.log("Volume bid:                " + price.volume_bid);
    console.log("Volume ask:                " + price.volume_ask);
    console.log("Trades:                    " + price.trades);
    console.log("Open interest:             " + price.open_interest);
    console.log("Open interest change:      " + price.open_interest_change);
    console.log("Next day open interest:    " + price.next_day_open_interest);
    console.log("Implied volatility:        " + price.implied_volatility);
    console.log("Implied volatility change: " + price.implied_volatility_change);
    console.log("Delta:                     " + price.delta);
  });
}, function(error) {
  console.error(error);
});
